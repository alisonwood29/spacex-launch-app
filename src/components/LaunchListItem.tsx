import React from 'react';

import '../styles/LaunchListItem.scss';

export interface LaunchListItemProps {
    missionName: string;
    number: number;
    rocketName: string;
    launchDate: string;
  }

export const LaunchListItem: React.FC<LaunchListItemProps> = props => {

    return (
        <div className='launch-list-item'>
            <div className='launch-item-left'>
                <div className='launch-number'>#{props.number}</div> 
                <div className='mission-name' data-testid='mission-name'>{props.missionName}</div>
            </div>
            <div className='launch-item-right'>
                <div className='launch-date'>{props.launchDate} </div>
                <div className='rocket-name'>{props.rocketName}</div>
            </div>
        </div>
    )
};