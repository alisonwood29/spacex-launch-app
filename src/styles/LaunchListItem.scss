@import './stylesheet.scss';

.launch-list-item {
    display: flex;
    justify-content: space-between;
    border-radius: 10px;
    padding: 12px 16px 12px 32px;
    box-shadow: 0px 2px 5px #bfbfbf;
    margin-bottom: 5px;

    .launch-item-left {
        display: flex;

        .launch-number {
        color: $font-color;
        font-size: 40px;
        font-weight: 500;
        width: 100px;
        font-family: $font-family;
        }

        .mission-name {
        color: $font-color;
        font-size: 32px;
        font-weight: 500;
        font-family: $font-family;
        }
    }

    .launch-item-right {
        display: flex;
        flex-direction: column;

        .launch-date {
            color: $font-color;
            font-size: 16px;
            font-weight: 400;
            font-family: $font-family;
        }

        .rocket-name {
            color: $font-color;
            font-size: 24px;
            font-weight: 700;
            font-family: $font-family;
        }
    }
}