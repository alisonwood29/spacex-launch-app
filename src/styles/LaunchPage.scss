.launch-page {
    display: flex;
    padding-left: 40px;
    padding-right: 40px;


    .spacex-logo {
        width: 280px;
        height: 31px;
        margin-top: 32px;
    }

    .launch-logo {
        width: 533.54px;
        height: 694px;
        margin-top: 138px;
        margin-right: 60px;
    }
}