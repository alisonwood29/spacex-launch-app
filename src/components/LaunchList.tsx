import React from 'react';
import { format } from 'date-fns';

import { LaunchListItem } from './LaunchListItem';
import { LaunchItem } from '../pages/LaunchPage';

import '../styles/LaunchList.scss';

export interface LaunchListProps {
    launches: Array<LaunchItem>;
};

export const LaunchList: React.FC<LaunchListProps> = props => {

    const formatDate: (date: Date) => string = date => {
        return format(date, 'dd LLL yyyy');
      };

    return (
        <div className='launch-list' data-testid='launch-list'>
           {props.launches.map((launch: LaunchItem, index: number) => {
                return (
                    <LaunchListItem 
                        missionName={launch.missionName}
                        number={index+1}
                        rocketName={launch.rocketName}
                        launchDate={formatDate(launch.launchDate)}
                        key={index}
                    />
                    );
                })
            }
        </div>
    )
};