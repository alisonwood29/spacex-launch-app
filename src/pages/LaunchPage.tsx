import React, {useState, useEffect} from 'react';

import {LaunchList} from '../components/LaunchList';
import {getAllLaunches} from '../api';

import '../styles/LaunchPage.scss';
import spacexLogo from '../assets/img/spacex-logo.png';
import launchLogo from '../assets/img/launch-home.png';

export interface LaunchItem {
    missionName: string;
    rocketName: string;
    launchDate: Date;
}

export const LaunchPage = () => {

    const [launches, setLaunches] = useState<Array<LaunchItem>>([]);

    const loadLaunches = async () => {
        const apiLaunches = await getAllLaunches();
        let launchItems: LaunchItem[] = new Array<LaunchItem>();
        apiLaunches.forEach((launch: any) => {
            const newLaunchItem: LaunchItem = {
                missionName: launch.mission_name,
                rocketName: launch.rocket.rocket_name,
                launchDate: new Date(launch.launch_date_local)
              };
              launchItems.push(newLaunchItem);
        });
        setLaunches(launchItems);
    };

    useEffect(() => {
        loadLaunches();
    }, []);

    return (
        <div className='launch-page'>
            <div className='launch-page-header'>
                <img className='spacex-logo' src={spacexLogo} alt='rocket-launch'/>
                <button
                onClick={() => loadLaunches()}>
                    Reload data
                    </button>
            </div>
            <div className='launch-page-content'>
                <img className='launch-logo' src={launchLogo} alt='rocket-launch'/>
                <LaunchList launches={launches}/>
           </div>
        </div>
    )
};