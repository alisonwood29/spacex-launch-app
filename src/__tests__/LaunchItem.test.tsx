import { render, getByTestId } from '@testing-library/react';

import { LaunchListItem } from '../components/LaunchListItem';
import { LaunchItem } from '../pages/LaunchPage';

const mockProps = {
    number: 1,
    missionName: 'Mission 1',
    rocketName: 'Rocket 1',
    launchDate: '25 March 2006'
};

describe('Launch Item Component tests', () => {
  it('renders', () => {
    const { container } = render(<LaunchListItem {...mockProps} />);

    expect(container).toBeDefined();
    expect(container).not.toBeNull();
  });

  it('renders a snapshot', () => {
    const { container } = render(<LaunchListItem {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});