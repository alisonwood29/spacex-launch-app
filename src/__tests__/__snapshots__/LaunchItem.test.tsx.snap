// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Item Component tests renders a snapshot 1`] = `
<div>
  <div
    class="launch-list-item"
    data-testid="launch-list-item"
  >
    <div
      class="launch-item-left"
    >
      <div
        class="launch-number"
      >
        #
        1
      </div>
      <div
        class="mission-name"
        data-testid="mission-name"
      >
        Mission 1
      </div>
    </div>
    <div
      class="launch-item-right"
    >
      <div
        class="launch-date"
      >
        25 March 2006
         
      </div>
      <div
        class="rocket-name"
      >
        Rocket 1
      </div>
    </div>
  </div>
</div>
`;
